tests/models/test_model_input_examples.py

```diff
@@ -305,3 +307,89 @@ def predict(self, X):
             AnyStringWith("Failed to infer the model signature from the input example."),
             AnyStringWith("oh no!"),
         )
+
+
+@pytest.fixture(scope="module")
+def iris_model():
+    X, y = datasets.load_iris(return_X_y=True, as_frame=True)
+    knn_model = knn.KNeighborsClassifier()
+    knn_model.fit(X, y)
+    return knn_model
+
+
+@pytest.mark.parametrize(
+    "input_example",
+    [
+        {
+            "sepal length (cm)": 5.1,
+            "sepal width (cm)": 3.5,
+            "petal length (cm)": 1.4,
+            "petal width (cm)": 0.2,
+        },
+        [5.1, 3.5, 1.4, 0.2],
+        pd.DataFrame(
+            {
+                "sepal length (cm)": 5.1,
+                "sepal width (cm)": 3.5,
+                "petal length (cm)": 1.4,
+                "petal width (cm)": 0.2,
+            },
+            index=[0],
+        ),
+    ],
+)
+def test_infer_signature_on_multi_column_input_examples(input_example, iris_model):
+    artifact_path = "model"
+
+    with mlflow.start_run():
+        mlflow.sklearn.log_model(
+            iris_model, artifact_path=artifact_path, input_example=input_example
+        )
+        model_uri = mlflow.get_artifact_uri(artifact_path)
+
+    mlflow_model = Model.load(model_uri)
+    input_columns = mlflow_model.signature.inputs.inputs
+    assert len(input_columns) == 4
+    for col in input_columns:
+        assert col.type == DataType.double
+    assert mlflow_model.signature.outputs == Schema([ColSpec(type=DataType.long)])
+
+
+@pytest.mark.parametrize(
+    "input_example",
+    ["some string", bytes([1, 2, 3])],
+)
+def test_infer_signature_on_scalar_input_examples(input_example):
+    class IdentitySklearnModel(BaseEstimator, ClassifierMixin):
+        def fit(self, X, y=None):
+            return self
+
+        def predict(self, X):
+            if isinstance(X, pd.DataFrame):
+                return X
+            raise Exception("Unsupported input type")
+
+    artifact_path = "model"
+
+    with mlflow.start_run():
+        mlflow.sklearn.log_model(
+            IdentitySklearnModel(), artifact_path=artifact_path, input_example=input_example
+        )
+        model_uri = mlflow.get_artifact_uri(artifact_path)
+
+    mlflow_model = Model.load(model_uri)
+    signature = mlflow_model.signature
+    assert isinstance(signature, ModelSignature)
+    assert signature.inputs.inputs[0].name == 0
+    if isinstance(input_example, str):
+        assert signature == ModelSignature(
+            inputs=Schema([ColSpec(name=0, type=DataType.string)]),
+            outputs=Schema([ColSpec(name=0, type=DataType.string)]),
+        )
+    else:
+        assert signature == ModelSignature(
+            inputs=Schema([ColSpec(name=0, type=DataType.binary)]),
+            outputs=Schema([ColSpec(name=0, type=DataType.binary)]),
+        )
```

Comment:
/ai simplify this code by only changing the value passed to the `type` argument depending on the `input_example` type.

Reply:
```suggestion
t = DataType.string if isinstance(input_example, str) else DataType.binary)
assert signature == ModelSignature(
    inputs=Schema([ColSpec(name=0, type=t]),
    outputs=Schema([ColSpec(name=0, type=t]),
)
```